en/tutorials/rag-blueprint.md

```diff
@@ -0,0 +1,1475 @@
+
+With Vespa, it is now possible to return only the top k most relevant chunks of a document, and include and combine both document-level and chunk-level features in ranking. 
+
+### Chunk selection
+
+You may not want to feed every chunk of the top k documents to the LLM.
+
+1. Use all chunks - will lead to a lot of irrelevant context, and potentially performance issues by returning too many chunks.
+2. Score and select chunks in your frontend - fixes irrelevant context, but adds complexity and does not address performance issues.
+3. Score and select chunks in the search-engine - allows you to return only the most relevant chunks. Best of both worlds.
+
+Below, we show how we can score and select the best chunks in Vespa.
+
+Compute closeness per chunk in a ranking function; use `elementwise(bm25(chunks), i, double)` for a per-chunk text signal.
+Now available: elementwise rank functions and filtering on the content nodes.
+
+You can pick a large document as the searchable unit, while still addressing the potential drawbacks many encounter as follows:
+
+* Pick your (larger) document as your searchable unit.
+* Chunk the text-fields automatically on indexing.
+* Embed each chunk (enabled through Vespa's multivector support)
+* Calculate chunk-level features (e.g. bm25 and embedding similarity) and document-level features. Combine as you want.
+* Limit the actual chunks that are returned to the ones that are actually relevant context for the LLM.
+
+**Conclusion:** It's often better to err on the side of slightly larger units.
+
+* LLMs are increasingly capable of handling larger contexts.
+* In Vespa, you can index larger units, while avoiding data duplication and performance issues, by returning only the most relevant parts.
+
+Vespa supports automatic [chunking](../reference/indexing-language-reference.html#converters) in the [indexing language](../indexing.html).
+
+Here is the parts of the schema, which defines the searchable unit as a document with a text field, and automatically chunks it into smaller parts of 1024 characters, which each are embedded and indexed separately:
+
+```txt
+field chunks type array<string> {
+    indexing: input text | chunk fixed-length 1024 | summary | index
+    index: enable-bm25
+}
+
+field chunk_embeddings type tensor<int8>(chunk{}, x[96]) {
+    indexing: input text | chunk fixed-length 1024 | embed | pack_bits | attribute | index
+    attribute {
+        distance-metric: hamming
+    }
+}
+```
+
+In Vespa, we can specify which chunks to be returned with a summary feature, see [docs](../reference/schema-reference.html#select-elements-by) for details. For this blueprint, we will return the top 3 chunks based on the similarity score of the chunk embeddings, which is calculated in the ranking phase. Note that this feature could be any chunk-level summary feature defined in your rank-profile.
+
+Here is how the summary feature is calculated in the rank-profile:
+
+```txt
+function chunk_emb_vecs() {
```

Comment:
I think it would be nice to have some comments here on what each function/expression does.